frontend/src/App.tsx

```diff
@@ -6,19 +6,42 @@ import TodoList from "./components/TodoList";
 import { Todo } from "./models/todo.model";
 
 function App() {
-  const [todos, setTodos] = useState<Todo[]>([
-    { description: "Hello world", id: "244454", isDone: false },
-    { description: "Hello Pakistan", id: "24454", isDone: false },
-    { description: "Hello GB", id: "24433", isDone: false },
-  ]);
-  const [newTodo, setNewTodo] = useState<Todo>({
-    description: "Hello world",
-    id: "244454",
-    isDone: false,
-  });
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodoDescription, setNewTodoDescription] = useState("");
 
   const onChange = (name: string, value: string) => {
-    console.log(name, ":", value);
+    setNewTodoDescription(value);
+  };
+
+  async function createTodo(url: string, description: string) {
+    return fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ description: description }),
+    });
+  }
+
+  const addTask = async () => {
+    if (newTodoDescription.trim().length === 0) {
```

Comment:
You are trimming the `description` here, but still sending the entered `description` in the request. It would be better to store the trimmed `description` in a local variable, and use that to both check the `length` and send in the `createTodo` function.